pkgs/development/python-modules/taichi/default.nix

```diff
@@ -0,0 +1,250 @@
+    # rm -rf ./tmp
+  '';
+
+  taichi =
+    # let
+    # in
+    # If the flag is a boolean, we'll pass "ON" or "OFF" to CMake.
+    # Otherwise, we'll pass the value of the flag.
+    stdenv.mkDerivation {
+      inherit
+        pname
+        version
+        src
+        cmakeFlags
+        preConfigure
+        ;
+
+      propagatedNativeBuildInputs = [
+        autoPatchelfHook
+        cmake
+        pybind11
+        git
+      ] ++ lib.optionals cudaSupport [ autoAddDriverRunpath ];
+
+      propagatedBuildInputs = [
+        elfutils
+        libbfd
+        libdwarf
+        libdwarf.dev
+        libffi.dev
+        libxml2.dev
+
+        libX11
+        libXrandr
+        libXinerama
+        libXcursor
+        libXi
+
+        libllvm
+      ] ++ platformLibs;
+
+      # See https://github.com/NixOS/nixpkgs/issues/144170
+      # Taken from original PR,
+      # maybe this is more future proof:
+      # https://github.com/NixOS/nixpkgs/issues/144170#issuecomment-1423195220
+      postPatch = ''
+        substituteInPlace \
+          external/SPIRV-Headers/SPIRV-Headers.pc.in \
+          external/SPIRV-Cross/pkg-config/spirv-cross-c-shared.pc.in \
+          external/SPIRV-Tools/cmake/SPIRV-Tools.pc.in \
+          external/SPIRV-Tools/cmake/SPIRV-Tools-shared.pc.in \
+          --replace-warn '$'{prefix}/@CMAKE_INSTALL_LIBDIR@ @CMAKE_INSTALL_FULL_LIBDIR@  \
+          --replace-warn '$'{prefix}/@CMAKE_INSTALL_INCLUDEDIR@ @CMAKE_INSTALL_FULL_INCLUDEDIR@
+      '';
+
+      doCheck = true;
+      nativeCheckInputs = [ gtest ];
+    };
+in
+buildPythonPackage {
+  inherit
+    pname
+    version
+    src
+    cmakeFlags
+    stdenv
+    preConfigure
+    ;
+
+  nativeBuildInputs =
+    [
+      ninja
+      numpy
+      scikit-build
+      setuptools
+      wheel
+    ]
+    ++ lib.optionals (stdenv.hostPlatform == stdenv.buildPlatform) [
+      pythonImportsCheckHook
+      pythonCatchConflictsHook
+    ];
+
+  buildInputs = [
+    taichi
+  ];
+
+  dependencies = [
+    colorama
+    dill
+    matplotlib
+    numpy
+    rich
+  ];
+
+  postConfigure = ''
+    cd ..
+  '';
+
+  # This is needed for the libraries like opengl and vulkan to be found at runtime
+  preInstallCheck = ''
+    shopt -s globstar
+
+    for file in $out/**/*.so; do
+      patchelf --add-rpath "${lib.makeLibraryPath platformLibs}" "$file"
+    done
+  '';
+
+  # dontUseCmakeConfigure = true;
+  # dontUseNinjaBuild = true;
+  # dontUseNinjaInstall = true;
+  dontUseNinjaCheck = true;
+
+  doCheck = true;
+  nativeCheckInputs = [ pytest ];
+
+  pythonImportsCheck = [
+    "taichi"
+  ];
+
+  meta = with lib; {
+    description = "Productive & portable high-performance programming in Python";
+    homepage = "https://github.com/taichi-dev/taichi";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ arunoruto ];
+    platforms = [
+      "x86_64-linux"
+      "x86_64-darwin"
+      "aarch64-darwin"
+    ];
+    mainProgram = "ti";
```

Comment:
try to add a versionCheckHook

Reply:
Thanks for the tip! Didn't know about this :)